### ISSUE TITLE
Titles are 403 forbidden

### ISSUE THREAD

**Issue by chrisjrob** [chrisjrob/sluggle#21]
```
https://www.google.co.uk/search?q=chav
http://tinyurl.com/hc6l7x4 - 403 Forbidden.
```

```
https://www.google.co.uk/?q=chav
http://tinyurl.com/j2mgqex - Google.
```

```
https://www.google.co.uk/#q=chav
http://tinyurl.com/j2mgqex - Google.
```

**Comment by chrisjrob** [comment]
Seems Google redirect `search?q=` to `#q=` to presumable the former is outdated. No neat way of fixing without a hacky regex substitution. These requests will probably die out over time.

**Comment by chrisjrob** [comment]
Doesn't just happen with Google search links. See also:

http://www.coindesk.com/making-sense-bitcoins-halving/

https://www.scan.co.uk/info/royalwarrant

**Comment by chrisjrob** [comment]
grifferz wonders if this sites blocking the default LWP user agent. Worth testing.

**Comment by domcleal** [comment]
I think it is, e.g.

<pre>
$ curl -v -o /dev/null -H "User-Agent: libwww-perl/6.13" "https://www.scan.co.uk/info/royalwarrant"
* Connected to www.scan.co.uk (104.16.85.28) port 443 (#0)
[...]
> GET /info/royalwarrant HTTP/1.1
> User-Agent: libwww-perl/6.13
> 
< HTTP/1.1 403 Forbidden
< Date: Tue, 14 Jun 2016 10:22:00 GMT
< Content-Type: text/html; charset=UTF-8
< Transfer-Encoding: chunked
< Connection: keep-alive
< Set-Cookie: __cfduid=db410d43e57f1ce9985b63a179a07f9201465899720; expires=Wed, 14-Jun-17 10:22:00 GMT; path=/; domain=.scan.co.uk; HttpOnly
< Cache-Control: max-age=10
< Expires: Tue, 14 Jun 2016 10:22:10 GMT
< X-Frame-Options: SAMEORIGIN
< Server: cloudflare-nginx
< CF-RAY: 2b2d0f05d57035c6-LHR
</pre>

<pre>
$ curl -v -o /dev/null "https://www.scan.co.uk/info/royalwarrant"
[...]
> GET /info/royalwarrant HTTP/1.1
> User-Agent: curl/7.40.0
> 
< HTTP/1.1 200 OK
[...]
</pre>

**Comment by domcleal** [comment]
I think all of the affected URLs are behind cloudflare, so blame chris_.

**Comment by chrisjrob** [comment]
The following site still not working:
https://www.helpscout.net/customer-acquisition

**Comment by chrisjrob** [comment]
This site does not appear to be a useragent issue, at least I cannot find a useragent that works. also error is Title Not Found, rather than forbidden. Seems LWP may not be able to get this site at all, as it reports success but does not actually have the title.

**Comment by chrisjrob** [comment]
headers returned look like this:

```
Cache-Control: max-age=604800
Connection: close
Date: Tue, 14 Jun 2016 11:14:27 GMT
Via: 1.1 c66d7e31512bf340e32266eebd70b64a.cloudfront.net (CloudFront)
Age: 23718
ETag: "2f9555de15031f57e89664fa8469ebc9"
Server: AmazonS3
Content-Encoding: gzip
Content-Length: 15105
Content-Type: text/html; charset=utf-8
Last-Modified: Mon, 13 Jun 2016 13:59:23 GMT
Client-Date: Tue, 14 Jun 2016 17:49:44 GMT
Client-Peer: 52.85.69.219:443
Client-Response-Num: 1
Client-SSL-Cert-Issuer: /C=US/O=DigiCert Inc/OU=www.digicert.com/CN=DigiCert SHA2 Extended Validation Server CA
Client-SSL-Cert-Subject: /businessCategory=Private Organization/1.3.6.1.4.1.311.60.2.1.3=US/1.3.6.1.4.1.311.60.2.1.2=Delaware/serialNumber=5003356/street=3rd Floor/street=131 Tremont Street/postalCode=02111/C=US/ST=Massachusetts/L=Boston/O=Help Scout Inc./CN=www.helpscout.net
Client-SSL-Cipher: ECDHE-RSA-AES128-GCM-SHA256
Client-SSL-Socket-Class: IO::Socket::SSL
X-Amz-Cf-Id: 13bXOGds62J2WXkaMPm9hBz3XkWcbt5qzGuShcrfXDvRv9mcI_s0Nw==
X-Cache: Hit from cloudfront
```

There is no Title: - and yet if you look at the page source it exists!

**Comment by grifferz** [comment]
Is it delivering you actual different content depending on your user agent? i.e. are you saying "if you look at the source" from your web browser's "show source"?

**Comment by chrisjrob** [comment]
No, this seems to be unrelated to the user agent issue, I should probably open a new issue. All I've been able to work out so far is that the web source has title tags (when you view source in the web browser), but the headers being reported from LWP::UserAgent do not include title (see above).

The other thing I note is that the content is gzipped, which may possibly be related.

All very odd. It may be peculiar to this single website!

**Comment by chrisjrob** [comment]
Last example of Forbidden was on 13 June, so seems to be fixed. Think that particular example was a freak.
